daemon/images/image_search_test.go

```diff
@@ -337,38 +332,17 @@ func TestSearchRegistryForImages(t *testing.T) {
 			},
 		},
 	}
-	for index, s := range successCases {
+	for _, tc := range successCases {
```

Comment:
Might be worth introducing subtests here too.

Reply:
Good one; looking at those tests, I actually thought I did _just that_, but maybe that got into another branch I was working on :joy:

I don't recall which branch that was in, so I just spend a few minutes to make the changes again in this one :shrug:

Now looks like;

```bash
go test -v -run TestSearchRegistryForImages ./daemon/images/
=== RUN   TestSearchRegistryForImagesErrors
=== RUN   TestSearchRegistryForImagesErrors/search_unknown_error
=== RUN   TestSearchRegistryForImagesErrors/invalid_filter_'type'
=== RUN   TestSearchRegistryForImagesErrors/invalid_filter_'is-automated=[invalid]'
=== RUN   TestSearchRegistryForImagesErrors/invalid_filter_'is-automated
=== RUN   TestSearchRegistryForImagesErrors/invalid_filter_'is-official=[invalid]'
=== RUN   TestSearchRegistryForImagesErrors/invalid_filter_'is-official
=== RUN   TestSearchRegistryForImagesErrors/invalid_filter_'stars=invalid'
=== RUN   TestSearchRegistryForImagesErrors/invalid_filter_'stars=invalid'#01
--- PASS: TestSearchRegistryForImagesErrors (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/search_unknown_error (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/invalid_filter_'type' (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/invalid_filter_'is-automated=[invalid]' (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/invalid_filter_'is-automated (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/invalid_filter_'is-official=[invalid]' (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/invalid_filter_'is-official (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/invalid_filter_'stars=invalid' (0.00s)
    --- PASS: TestSearchRegistryForImagesErrors/invalid_filter_'stars=invalid'#01 (0.00s)
=== RUN   TestSearchRegistryForImages
=== RUN   TestSearchRegistryForImages/empty_results
=== RUN   TestSearchRegistryForImages/no_filter
=== RUN   TestSearchRegistryForImages/is-automated=true,_no_results
=== RUN   TestSearchRegistryForImages/is-automated=true
=== RUN   TestSearchRegistryForImages/is-automated=false,_no_results
=== RUN   TestSearchRegistryForImages/is-automated=false
=== RUN   TestSearchRegistryForImages/is-official=true,_no_results
=== RUN   TestSearchRegistryForImages/is-official=true
=== RUN   TestSearchRegistryForImages/is-official=false,_no_results
=== RUN   TestSearchRegistryForImages/is-official=false
=== RUN   TestSearchRegistryForImages/stars=0
=== RUN   TestSearchRegistryForImages/stars=0,_no_results
=== RUN   TestSearchRegistryForImages/stars=1
=== RUN   TestSearchRegistryForImages/stars=1,_is-official=true,_is-automated=true
--- PASS: TestSearchRegistryForImages (0.00s)
    --- PASS: TestSearchRegistryForImages/empty_results (0.00s)
    --- PASS: TestSearchRegistryForImages/no_filter (0.00s)
    --- PASS: TestSearchRegistryForImages/is-automated=true,_no_results (0.00s)
    --- PASS: TestSearchRegistryForImages/is-automated=true (0.00s)
    --- PASS: TestSearchRegistryForImages/is-automated=false,_no_results (0.00s)
    --- PASS: TestSearchRegistryForImages/is-automated=false (0.00s)
    --- PASS: TestSearchRegistryForImages/is-official=true,_no_results (0.00s)
    --- PASS: TestSearchRegistryForImages/is-official=true (0.00s)
    --- PASS: TestSearchRegistryForImages/is-official=false,_no_results (0.00s)
    --- PASS: TestSearchRegistryForImages/is-official=false (0.00s)
    --- PASS: TestSearchRegistryForImages/stars=0 (0.00s)
    --- PASS: TestSearchRegistryForImages/stars=0,_no_results (0.00s)
    --- PASS: TestSearchRegistryForImages/stars=1 (0.00s)
    --- PASS: TestSearchRegistryForImages/stars=1,_is-official=true,_is-automated=true (0.00s)
PASS
ok  	github.com/docker/docker/daemon/images	0.036s
```